src/cluster/replication.cc

```diff
@@ -732,6 +732,13 @@ ReplicationThread::CBState ReplicationThread::fullSyncReadCB(bufferevent *bev) {
       LOG(INFO) << "[replication] Succeeded restoring the backup, fullsync was finish";
       post_fullsync_cb_();
 
+      // It needs to reload namespaces from DB after the full sync is done,
+      // or namespaces are not visible in the replica.
+      s = srv_->GetNamespace()->LoadAndRewrite();
+      if (!s.IsOK()) {
+        LOG(ERROR) << "[replication] Failed to load and rewrite namespace: " << s.Msg();
+      }
+
```

Comment:
Extract a `PostFullSync` function, move it to this function?

Reply:
We have `post_fullsync_cb` now, but it's used for cleaning resources if failed to do the full sync. So it should be not good to put this inside that.

Reply:
Isn't it a callback function after successfully loading the data?

https://github.com/apache/kvrocks/blob/2836bf02877fa9853af6047dc35b206dea7330f6/src/server/server.cc#L284-L290

Reply:
No indeed, it should be used for resetting resources corresponding to `pre_fullsync_cb`. That said, it must be called once if the `pre_fullsync_cb` was invoked.